src/quemb/molbe/mbe.py

```diff
@@ -172,9 +178,18 @@ def __init__(
         self.pot = initialize_pot(self.fobj.Nfrag, self.fobj.edge_idx)
 
         if scratch_dir is None:
-            self.scratch_dir = WorkDir.from_environment()
-        else:
+            self.scratch_dir = WorkDir.from_environment(
+                cleanup_at_end=self.cleanup_at_end,
+            )
+        elif isinstance(scratch_dir, WorkDir):
             self.scratch_dir = scratch_dir
+
+        elif isinstance(scratch_dir, PathLike):
+            self.scratch_dir = WorkDir(
+                path=Path(scratch_dir),
```

Comment:
The cast to Path is redundant here. **but** this comment is anyway superfluos, since this elif branch should IMHO not exist.